src/docbuilder/rustwide_builder.rs

```diff
@@ -1160,4 +1160,56 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    #[ignore]
+    fn test_implicit_features_for_optional_dependencies() {
```

Comment:
While these specific tests could be left here as integration tests, 

IMO the general get-features-from-metadata should be tested on `get_features` directly, passing in some static metadata. 

Or am I missing something?